diff -r -u ../../../vpp-superdesk-core/superdesk/metadata/item.py ./metadata/item.py
--- ../../../vpp-superdesk-core/superdesk/metadata/item.py	2015-10-15 02:03:26.052494434 +0200
+++ ./metadata/item.py	2015-10-15 02:16:19.551522558 +0200
@@ -270,6 +270,50 @@
     'contents': {
         'type': 'list'
     },
+    'verification': {
+        'type': 'dict',
+        'nullable': True,
+        'schema': {
+            'results': Resource.rel('verification_results'),
+            'stats': {
+                'type': 'dict',
+                'nullable': True,
+                'schema': {
+                    'izitru': {
+                        'type': 'dict',
+                        'nullable': True,
+                        'schema': {
+                            'verdict': {'type': 'integer'},
+                            'location': {'type': 'string'},
+                        },
+                    },
+                    'tineye': {
+                        'type': 'dict',
+                        'nullable': True,
+                        'schema': {
+                            'total': {'type': 'integer'},
+                        },
+                    },
+                    'incandescent': {
+                        'type': 'dict',
+                        'nullable': True,
+                        'schema': {
+                            'total_google': {'type': 'integer'},
+                            'total_bing': {'type': 'integer'},
+                            'total_baidu': {'type': 'integer'},
+                            'total_yandex': {'type': 'integer'},
+                            'total_other': {'type': 'integer'},
+                        },
+                    },
+                },
+            }
+        }
+    },
+    'vpp_tag': {
+        'type': 'list',
+        'nullable': True,
+        'schema': {'type': 'string'},
+    },
 
     # aka Locator as per NewML Specification
     'place': {
@@ -324,7 +368,12 @@
         'type': 'datetime',
         'versioned': False
     },
-    'lock_session': Resource.rel('auth')
+    'lock_session': Resource.rel('auth'),
+
+    'viewed': {
+        'type': 'boolean',
+        'default': False,
+    },
 }
 
 metadata_schema['lock_user']['versioned'] = False
Only in ./metadata/: __pycache__
diff -r -u ../../../vpp-superdesk-core/superdesk/metadata/utils.py ./metadata/utils.py
--- ../../../vpp-superdesk-core/superdesk/metadata/utils.py	2015-10-15 02:03:26.052494434 +0200
+++ ./metadata/utils.py	2015-10-15 02:16:31.798702210 +0200
@@ -30,6 +30,11 @@
     'day': {'date_range': {'field': 'firstcreated', 'format': 'dd-MM-yyy HH:mm:ss', 'ranges': [{'from': 'now-24H'}]}},
     'week': {'date_range': {'field': 'firstcreated', 'format': 'dd-MM-yyy HH:mm:ss', 'ranges': [{'from': 'now-1w'}]}},
     'month': {'date_range': {'field': 'firstcreated', 'format': 'dd-MM-yyy HH:mm:ss', 'ranges': [{'from': 'now-1M'}]}},
+    'make': {'terms': {'field': 'filemeta.Make'}},
+    'capture_location': {'terms': {'field': 'verification.stats.izitru.location'}},
+    'izitru': {'terms': {'field': 'verification.stats.izitru.verdict'}},
+    'original_source': {'terms': {'field': 'original_source'}},
+    'vpp_tag': {'terms': {'field': 'vpp_tag'}},
 }
 
 
